openwisp_monitoring/check/classes/iperf.py

```diff
@@ -115,21 +186,71 @@ def check(self, store=True):
             org_id = str(self.related_object.organization.id)
             self.validate_params(params=iperf_config[org_id])
 
+        username = self._get_param('username', 'username.default')
         port = self._get_param(
             'client_options.port', 'client_options.properties.port.default'
         )
         time = self._get_param(
             'client_options.time', 'client_options.properties.time.default'
         )
+        bytes = self._get_param(
+            'client_options.bytes', 'client_options.properties.bytes.default'
+        )
+        blockcount = self._get_param(
+            'client_options.blockcount', 'client_options.properties.blockcount.default'
+        )
+        window = self._get_param(
+            'client_options.window', 'client_options.properties.window.default'
+        )
+        parallel = self._get_param(
+            'client_options.parallel', 'client_options.properties.parallel.default'
+        )
+        reverse = self._get_param(
+            'client_options.reverse', 'client_options.properties.reverse.default'
+        )
+        bidirectional = self._get_param(
+            'client_options.bidirectional',
+            'client_options.properties.bidirectional.default',
+        )
+        ct = self._get_param(
+            'client_options.connect_timeout',
+            'client_options.properties.connect_timeout.default',
+        )
         tcp_bitrate = self._get_param(
             'client_options.tcp.bitrate',
             'client_options.properties.tcp.properties.bitrate.default',
         )
+        tcp_length = self._get_param(
+            'client_options.tcp.length',
+            'client_options.properties.tcp.properties.length.default',
+        )
         udp_bitrate = self._get_param(
             'client_options.udp.bitrate',
             'client_options.properties.udp.properties.bitrate.default',
         )
-        username = self._get_param('username', 'username.default')
+        udp_length = self._get_param(
+            'client_options.udp.length',
+            'client_options.properties.udp.properties.length.default',
+        )
+
+        # by default we use 'time' param
+        # for the iperf test end condition
+        test_end_condition = f'-t {time}'
+        # if 'bytes' present in config
+        # use it instead of 'time'
+        if bytes:
+            test_end_condition = f'-n {bytes}'
+        # if 'blockcount' present in config
+        # use it instead of 'time' or 'bytes'
+        if blockcount:
+            test_end_condition = f'-k {blockcount}'
+        # only one reverse condition can be use
+        # reverse or bidirectional not both
+        rev_or_bidir = ''
+        if reverse:
+            rev_or_bidir = '--reverse'
+        if bidirectional:
+            rev_or_bidir = '--bidir'
```

Comment:
I think we should move this to a separate method which returns the two commands.

Reply:
yes, thanks!